core/src/test/java/org/ehcache/EhcacheBasicRemoveAllEventsTest.java

```diff
@@ -0,0 +1,208 @@
+import org.ehcache.event.CacheEventListener;
+import org.ehcache.events.CacheEventNotificationService;
+import org.ehcache.events.CacheEventNotificationServiceImpl;
+import org.ehcache.exceptions.BulkCacheWritingException;
+import org.ehcache.exceptions.CacheAccessException;
+import org.ehcache.function.Function;
+import org.ehcache.spi.loaderwriter.CacheLoaderWriter;
+import org.ehcache.util.IsRemoved;
+import org.hamcrest.CoreMatchers;
+import org.junit.Test;
+import org.mockito.Matchers;
+import org.mockito.Mock;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class provides the events based for REMOVE_ALL operations.
+ */
+public class EhcacheBasicRemoveAllEventsTest extends EhcacheBasicCrudBase {
+
+  @Mock
+  protected CacheEventListener<String, String> cacheEventListener;
+
+  protected CacheEventNotificationService<String, String> cacheEventNotificationService;
+
+  protected IsRemoved<String, String> isRemoved = new IsRemoved<String, String>();
+
+  @Mock
+  protected CacheLoaderWriter<String, String> cacheLoaderWriter;
+
+  @Test
+  public void testRemoveAllNull() throws Exception {
+    final Ehcache<String, String> ehcache = this.getEhcache(null);
+    try {
+      ehcache.removeAll(null);
+      fail();
+    } catch (NullPointerException e) {
+      // Expected
+    }
+    verify(cacheEventListener, never()).onEvent(Matchers.<CacheEvent<String, String>>any());
+    ehcache.getRuntimeConfiguration().deregisterCacheEventListener(cacheEventListener);
+  }
+
+  @Test
+  public void testRemoveAllNullKey() throws Exception {
+    final Map<String, String> originalStoreContent = getEntryMap(KEY_SET_A, KEY_SET_B);
+    final FakeStore fakeStore = new FakeStore(originalStoreContent);
+    this.store = spy(fakeStore);
+    final Set<String> keys = new LinkedHashSet<String>();
+    for (final String key : KEY_SET_A) {
+      keys.add(key);
+      if ("keyA2".equals(key)) {
+        keys.add(null);
+      }
+    }
+    final Ehcache<String, String> ehcache = this.getEhcache(null);
+    try {
+      ehcache.removeAll(keys);
+      fail();
+    } catch (NullPointerException e) {
+      // Expected
+    }
+    verify(cacheEventListener, never()).onEvent(Matchers.<CacheEvent<String, String>>any());
+    try {
+      ehcache.getRuntimeConfiguration().deregisterCacheEventListener(cacheEventListener);
+      fail();
+    } catch (UnsupportedOperationException e) {
+      //expected
+    }
+  }
+
+  @Test
+  public void testRemoveAll() throws Exception {
+    final Map<String, String> originalStoreContent = getEntryMap(KEY_SET_A, KEY_SET_B);
+    final FakeStore fakeStore = new FakeStore(originalStoreContent, Collections.singleton("keyA3"));
+    this.store = spy(fakeStore);
+    final Ehcache<String, String> ehcache = this.getEhcache(null);
+    final Set<String> contentUpdates = fanIn(KEY_SET_A, KEY_SET_C);
+    ehcache.removeAll(contentUpdates);
+    verify(cacheEventListener, times(5)).onEvent(argThat(isRemoved));
```

Comment:
same comment as above, in general, explaining the expectation rather than a number would improve the understanding of the tests